src/action/base/create_or_merge_nix_config.rs

```diff
@@ -0,0 +1,665 @@
+                    }
+                } else {
+                    merged_nix_config
+                        .settings_mut()
+                        .insert(pending_conf_name.to_owned(), pending_conf_value.to_owned());
+                }
+            }
+
+            if !unmergeable_config_names.is_empty() {
+                return Err(ActionError::Custom(Box::new(
+                    CreateOrMergeNixConfigError::UnmergeableConfig(
+                        unmergeable_config_names,
+                        this.path,
+                    ),
+                )));
+            }
+
+            if !merged_nix_config.settings().is_empty() {
+                this.nix_configs.merged_nix_config = merged_nix_config;
+                return Ok(StatefulAction::uncompleted(this));
+            }
+
+            tracing::debug!(
+                "Setting Nix configurations in `{}` already complete",
+                this.path.display()
+            );
+            return Ok(StatefulAction::completed(this));
+        } else {
+            let mut merged_nix_config = NixConfig::new();
+            for (pending_conf_name, pending_conf_value) in pending_nix_config.settings() {
+                merged_nix_config
+                    .settings_mut()
+                    .insert(pending_conf_name.to_owned(), pending_conf_value.to_owned());
+            }
+
+            if !merged_nix_config.settings().is_empty() {
+                this.nix_configs.merged_nix_config = merged_nix_config;
+            }
+        }
+
+        Ok(StatefulAction::uncompleted(this))
+    }
+}
+
+#[async_trait::async_trait]
+#[typetag::serde(name = "create_or_merge_nix_config")]
+impl Action for CreateOrMergeNixConfig {
+    fn tracing_synopsis(&self) -> String {
+        format!("Create or merge nix.conf file `{}`", self.path.display())
+    }
+
+    fn tracing_span(&self) -> Span {
+        let span = span!(
+            tracing::Level::DEBUG,
+            "create_or_merge_nix_config",
+            path = tracing::field::display(self.path.display()),
+            mode = tracing::field::display(format!("{:#o}", NIX_CONF_MODE)),
+            merged_nix_config = tracing::field::Empty,
+        );
+
+        if tracing::enabled!(tracing::Level::TRACE) {
+            span.record(
+                "merged_nix_config",
+                &self
+                    .nix_configs
+                    .merged_nix_config
+                    .settings()
+                    .iter()
+                    .map(|(k, v)| format!("{k}='{v}'"))
+                    .collect::<Vec<_>>()
+                    .join(" "),
+            );
+        }
+        span
+    }
+
+    fn execute_description(&self) -> Vec<ActionDescription> {
+        vec![ActionDescription::new(
+            self.tracing_synopsis(),
+            vec![format!(
+                "If {} already exists, we will attempt to merge the current settings with our settings; \
```

Comment:
I'm not sure the wording/styling of using "our" here matches the current plan explanation format:

```
* Create directory `/nix`
* Fetch `https://releases.nixos.org/nix/nix-2.13.2/nix-2.13.2-x86_64-linux.tar.xz` to `/nix/temp-install-dir`
* Create build users (UID 30000-30032) and group (GID 30000)
  The Nix daemon requires system users (and a group they share) which it can act as in order to build
  Create group `nixbld` (GID 30000)
  Create user `nixbld0` (UID 30000) in group `nixbld` (GID 30000)
  Create user `nixbld1` (UID 30001) in group `nixbld` (GID 30000)
  Create user `nixbld2` (UID 30002) in group `nixbld` (GID 30000)
  Create user `nixbld3` (UID 30003) in group `nixbld` (GID 30000)
  Create user `nixbld4` (UID 30004) in group `nixbld` (GID 30000)
  Create user `nixbld5` (UID 30005) in group `nixbld` (GID 30000)
  Create user `nixbld6` (UID 30006) in group `nixbld` (GID 30000)
  Create user `nixbld7` (UID 30007) in group `nixbld` (GID 30000)
  Create user `nixbld8` (UID 30008) in group `nixbld` (GID 30000)
  Create user `nixbld9` (UID 30009) in group `nixbld` (GID 30000)
  Create user `nixbld10` (UID 30010) in group `nixbld` (GID 30000)
  Create user `nixbld11` (UID 30011) in group `nixbld` (GID 30000)
  Create user `nixbld12` (UID 30012) in group `nixbld` (GID 30000)
  Create user `nixbld13` (UID 30013) in group `nixbld` (GID 30000)
  Create user `nixbld14` (UID 30014) in group `nixbld` (GID 30000)
  Create user `nixbld15` (UID 30015) in group `nixbld` (GID 30000)
  Create user `nixbld16` (UID 30016) in group `nixbld` (GID 30000)
  Create user `nixbld17` (UID 30017) in group `nixbld` (GID 30000)
  Create user `nixbld18` (UID 30018) in group `nixbld` (GID 30000)
  Create user `nixbld19` (UID 30019) in group `nixbld` (GID 30000)
  Create user `nixbld20` (UID 30020) in group `nixbld` (GID 30000)
  Create user `nixbld21` (UID 30021) in group `nixbld` (GID 30000)
  Create user `nixbld22` (UID 30022) in group `nixbld` (GID 30000)
  Create user `nixbld23` (UID 30023) in group `nixbld` (GID 30000)
  Create user `nixbld24` (UID 30024) in group `nixbld` (GID 30000)
  Create user `nixbld25` (UID 30025) in group `nixbld` (GID 30000)
  Create user `nixbld26` (UID 30026) in group `nixbld` (GID 30000)
  Create user `nixbld27` (UID 30027) in group `nixbld` (GID 30000)
  Create user `nixbld28` (UID 30028) in group `nixbld` (GID 30000)
  Create user `nixbld29` (UID 30029) in group `nixbld` (GID 30000)
  Create user `nixbld30` (UID 30030) in group `nixbld` (GID 30000)
  Create user `nixbld31` (UID 30031) in group `nixbld` (GID 30000)
  Add user `nixbld0` (UID 30000) to group `nixbld` (GID 30000)
  Add user `nixbld1` (UID 30001) to group `nixbld` (GID 30000)
  Add user `nixbld2` (UID 30002) to group `nixbld` (GID 30000)
  Add user `nixbld3` (UID 30003) to group `nixbld` (GID 30000)
  Add user `nixbld4` (UID 30004) to group `nixbld` (GID 30000)
  Add user `nixbld5` (UID 30005) to group `nixbld` (GID 30000)
  Add user `nixbld6` (UID 30006) to group `nixbld` (GID 30000)
  Add user `nixbld7` (UID 30007) to group `nixbld` (GID 30000)
  Add user `nixbld8` (UID 30008) to group `nixbld` (GID 30000)
  Add user `nixbld9` (UID 30009) to group `nixbld` (GID 30000)
  Add user `nixbld10` (UID 30010) to group `nixbld` (GID 30000)
  Add user `nixbld11` (UID 30011) to group `nixbld` (GID 30000)
  Add user `nixbld12` (UID 30012) to group `nixbld` (GID 30000)
  Add user `nixbld13` (UID 30013) to group `nixbld` (GID 30000)
  Add user `nixbld14` (UID 30014) to group `nixbld` (GID 30000)
  Add user `nixbld15` (UID 30015) to group `nixbld` (GID 30000)
  Add user `nixbld16` (UID 30016) to group `nixbld` (GID 30000)
  Add user `nixbld17` (UID 30017) to group `nixbld` (GID 30000)
  Add user `nixbld18` (UID 30018) to group `nixbld` (GID 30000)
  Add user `nixbld19` (UID 30019) to group `nixbld` (GID 30000)
  Add user `nixbld20` (UID 30020) to group `nixbld` (GID 30000)
  Add user `nixbld21` (UID 30021) to group `nixbld` (GID 30000)
  Add user `nixbld22` (UID 30022) to group `nixbld` (GID 30000)
  Add user `nixbld23` (UID 30023) to group `nixbld` (GID 30000)
  Add user `nixbld24` (UID 30024) to group `nixbld` (GID 30000)
  Add user `nixbld25` (UID 30025) to group `nixbld` (GID 30000)
  Add user `nixbld26` (UID 30026) to group `nixbld` (GID 30000)
  Add user `nixbld27` (UID 30027) to group `nixbld` (GID 30000)
  Add user `nixbld28` (UID 30028) to group `nixbld` (GID 30000)
  Add user `nixbld29` (UID 30029) to group `nixbld` (GID 30000)
  Add user `nixbld30` (UID 30030) to group `nixbld` (GID 30000)
  Add user `nixbld31` (UID 30031) to group `nixbld` (GID 30000)
* Create a directory tree in `/nix`
  Create directory `/nix/var`
  Create directory `/nix/var/log`
  Create directory `/nix/var/log/nix`
  Create directory `/nix/var/log/nix/drvs`
  Create directory `/nix/var/nix`
  Create directory `/nix/var/nix/db`
  Create directory `/nix/var/nix/gcroots`
  Create directory `/nix/var/nix/gcroots/per-user`
  Create directory `/nix/var/nix/profiles`
  Create directory `/nix/var/nix/profiles/per-user`
  Create directory `/nix/var/nix/temproots`
  Create directory `/nix/var/nix/userpool`
  Create directory `/nix/var/nix/daemon-socket`
* Move the downloaded Nix into `/nix`
  Nix is being downloaded to `/nix/temp-install-dir` and should be in `/nix`
* Setup the default Nix profile
* Place the Nix configuration in `/etc/nix/nix.conf`
  This file is read by the Nix daemon to set its configuration options at runtime.
* Place channel configuration at `/root/.nix-channels`
* Configure the shell profiles
  Update shell profiles to import Nix
* Configure Nix daemon related settings with systemd
  Run `systemd-tempfiles --create --prefix=/nix/var/nix`
  Run `systemctl link /nix/var/nix/profiles/default/lib/systemd/system/nix-daemon.service`
  Run `systemctl link /nix/var/nix/profiles/default/lib/systemd/system/nix-daemon.socket`
  Run `systemctl daemon-reload`
  Run `systemctl enable --now /nix/var/nix/profiles/default/lib/systemd/system/nix-daemon.socket`
```

Perhaps like "Update the `/etc/nix/nix.conf` settings $LIST_OF_SETTINGS`?"